python/README.md

```diff
@@ -4,11 +4,122 @@ This library provides a Pythonic API wrapper for the reference Arrow C++
 implementation, along with tools for interoperability with pandas, NumPy, and
 other traditional Python scientific computing packages.
 
-#### Development details
+### Development details
 
 This project is layered in two pieces:
 
 * pyarrow, a C++ library for easier interoperability between Arrow C++, NumPy,
   and pandas
 * Cython extensions and pure Python code under arrow/ which expose Arrow C++
-  and pyarrow to pure Python users
\ No newline at end of file
+  and pyarrow to pure Python users
+
+#### PyArrow Installation
+These are instructions on how to install PyArrow from scratch on Linux (assuming arrow is not yet installed)
+
+1. **g++ and gcc**
+
+  Make sure the latest versions of g++ and gcc are installed.
+  ```bash
+  sudo add-apt-repository ppa:ubuntu-toolchain-r/test
+  sudo apt-get update
+  sudo apt-get install gcc-4.9 g++-4.9
+  ```
+
+2. **cmake**
+  ```bash
+  sudo add-apt-repository ppa:george-edison55/cmake-3.x
+  sudo apt-get update
+  sudo apt-get install cmake
+  ```
+
+3. **boost**
+  ```bash
+  sudo apt-get install build-essential g++ python-dev autotools-dev libicu-dev build-essential libbz2-dev
+  wget -O boost_1_60_0.tar.gz http://sourceforge.net/projects/boost/files/boost/1.60.0/boost_1_60_0.tar.gz/download
+  tar xzvf boost_1_60_0.tar.gz
+  cd boost_1_60_0
+  ./bootstrap.sh --prefix=/usr/local
+  sudo ./b2 install
```

Comment:
Since `libboost-all-dev` is sufficient on recent (e.g. Debian 14.04 and higher), I don't think we need to include boost build instructions, but we should state the minimum boost version if there is one.